test/Polly.Core.Tests/Timeout/TimeoutResilienceStrategyTests.cs

```diff
@@ -224,7 +224,7 @@ public async Task Execute_EnsureCancellationTokenRegistrationNotExecutedOnSynchr
             },
             cts.Token);
         }
-        catch (OperationCanceledException)
+        catch (TaskCanceledException)
```

Comment:
Is this change necessary?

Reply:
In the .NET 8 branch at least, the tests were failing without this change.

Reply:
This is super odd, the `TaskCanceledException` just derives from `OperationCanceledException`

Reply:
Let me just revert it locally and double-check in case I got confused somewhere...